packages/fxa-settings/src/lib/context/implementations/url-context.ts

```diff
@@ -0,0 +1,48 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { Context } from "..";
+
+export class UrlContext implements Context {
+
+  get pathName() {
+    return this.window.location.pathname;
+  }
+
+  constructor(
+    public readonly window: Window
+  ) {
+  }
+
+  get(key: string) {
+    const params = new URLSearchParams(this.window.location.search);
+    const rawValue = params.get(key);
+    if (rawValue == null) {
+      return null;
+    }
+
+    return window.decodeURIComponent(rawValue);
+  }
+
+  set(key: string, val: any): void {
+    // @ts-ignore... https://developer.mozilla.org/en-US/docs/Web/API/History/pushState
+    const url = new URL(this.window.location);
+    url.searchParams.set(key, window.encodeURIComponent(val));
+
+    // Use push state to persist URL, but prevent page loads.
+    this.window.history.pushState({}, '', url);
```

Comment:
This hasn't been vetted yet, since I'd like some feedback before putting time into writing tests. 

I think it'll work, but maybe we want to treat this more like the storage context, were we store the state in private property and then sync it periodically. We might have to see how this evolves as we actually use the models. Personally I like the idea of not having to worry about when to `load` and `persist` temporary state.

Reply:
Oh, yes replaceState, good call 😅.